app/forms/recovery_code_form.rb

```diff
@@ -21,6 +21,8 @@ def submit
   attr_reader :user, :success
 
   def valid_recovery_code?
+    word_regexp = /\w{#{RandomPhrase::WORD_LENGTH}}/
+    return false unless code =~ /\A#{word_regexp} #{word_regexp} #{word_regexp} #{word_regexp}\Z/
```

Comment:
I looked at the existing tests in `spec/forms/recovery_code_form_spec.rb` and they seemed to cover this case already so I wasn't sure I could improve on them.